src/morphoOracle.ts

```diff
@@ -0,0 +1,102 @@
+import { experimental_createEffect, S } from "envio";
+
+type HistoricalPricePoint = {
+  x: number
+  y: number
+}
+
+interface HistoricalAssetPriceResponse {
+  data: {
+    assetByAddress: {
+      historicalPriceUsd: HistoricalPricePoint[]
+    } | null
+  }
+  errors?: any
+}
+
+// Define the schema for the effect output
+const getMorphoHistoricalPriceSchema = S.schema({
+  price: S.number,
+})
+
+// Infer the type from the schema
+type getMorphoHistoricalPrice = S.Infer<typeof getMorphoHistoricalPriceSchema>;
+
+export const getMorphoHistoricalPrice = experimental_createEffect(
+  {
+    name: "getMorphoHistoricalPrice",
+    input: {
+      assetAddress: S.string,
+      chainId: S.number,
+      timestamp: S.bigint,
+    },
```

Comment:
Nice! Again if you can push the .cache that would be awesome. 

You need to hit sync button on envio.dev.console to create that cache file